daemon/k8s_watcher.go

```diff
@@ -62,17 +63,25 @@ func k8sErrorHandler(e error) {
 	if e == nil {
 		return
 	}
-	// Omitting the 'connection refused' common messages
-	if strings.Contains(e.Error(), "connection refused") {
-		firstK8sErrorLogMessage.Do(func() {
-			// Reset the timer for the first message
-			log.Error(e)
-			k8sLogMessagesTimer.Reset(k8sErrLogTimeout)
-		})
+	errstr := e.Error()
+	k8sErrMsgMU.Lock()
+	// if the error message already exists in the map then in print it
+	// otherwise we create a new timer for that specific message in the
+	// k8sErrMsg map.
+	if t, ok := k8sErrMsg[errstr]; !ok {
+		// Omitting the 'connection refused' common messages and
+		// Failed to list NetworkPolicy for k8s < 1.7
+		if strings.Contains(errstr, "connection refused") ||
```

Comment:
This is better but we should check k8s version and omit it altogether if it has no chance of succeeding. I suggest we don't even start listening for NetworkPolicy for 1.7. Instead, in that case, a single warning message, in the beginning, should indicate that due to version, NetworkPolicy is not supported.

Once a minute still means  1440 duplicate log messages every day.